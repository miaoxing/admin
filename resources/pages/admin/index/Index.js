import React from "react";
import {Col, Row, Card, ListGroup} from '@miaoxing/bootstrap';
import color from 'color';
import {Page} from "@miaoxing/a-page";
import {
  Chart,
  Geom,
  Axis,
  Tooltip,
  Legend,
} from "bizcharts";
import api from '@miaoxing/api';
import {Box} from 'rebass';
import {cls, floatRight} from '@miaoxing/css';

export default class extends React.Component {
  state = {
    charts: [],
    stats: []
  }

  componentDidMount() {
    api.cur({loading: true}).then(ret => this.setState(ret));
  }

  render() {
    return <Page
      raw
      breadcrumb={false}
    >
      <Card>
        <Card.Header>
          数据统计
        </Card.Header>
        <Card.Body>
          <Row>
            {this.state.stats.map(stat => (
              <Col key={stat.title} lg={3} md={6} sm={12}>
                <Box
                  as={Card}
                  color="white"
                  sx={{
                    border: 0,
                    background: `linear-gradient(135deg, ${stat.color}, ${color(stat.color).lighten(0.2).string()})`,
                  }}
                >
                  <Box as={Card.Body} fontWeight="light">
                    <Box as={Card.Title} fontWeight="light">{stat.title}</Box>
                    <Box as={Card.Text} mb={0} fontSize="3em">{stat.value}</Box>
                    <Box as="i" className={stat.icon} sx={{
                      opacity: .15,
                      position: 'absolute',
                      top: '50%',
                      transform: 'translateY(-50%)',
                      right: '2rem',
                      fontSize: '5rem',
                    }}/>
                  </Box>
                </Box>
              </Col>
            ))}
          </Row>
        </Card.Body>
      </Card>

      <Box as={Card} mt={4}>
        <Card.Header>
          交易走势
        </Card.Header>
        <Card.Body>
          <Chart
            height={400}
            data={this.state.charts}
            scale={{
              date: {
                type: 'time',
                mask: 'MM-DD'
              }
            }}
            forceFit
          >
            <Legend/>
            <Axis name="date"/>
            <Axis name="value"/>
            <Tooltip
              crosshairs={{
                type: "y"
              }}
            />
            <Geom
              type="line"
              position="date*value"
              size={2}
              color={"name"}
              shape={"smooth"}
            />
            <Geom
              type="point"
              position="date*value"
              size={4}
              shape={"circle"}
              color={"name"}
              style={{
                stroke: "#fff",
                lineWidth: 1
              }}
            />
          </Chart>
        </Card.Body>
      </Box>

      <Box as={Card} mt={4}>
        <Card.Header>
          更新日志
          <a href="#" target="_blank" className={floatRight()}>
            更多{' '}<i className="fa fa-angle-right"/>
          </a>
        </Card.Header>
        <ListGroup variant="flush">
          <ListGroup.Item className={cls({borderBottom: 0})}>
            <a href="#" target="_blank">Cras justo odio</a>
            <Box display="inline" color="muted" sx={{float: 'right'}}>2020-01-01</Box>
          </ListGroup.Item>
          <ListGroup.Item className={cls({borderBottom: 0})}>
            <a href="#" target="_blank">Dapibus ac facilisis in</a>
            <Box display="inline" color="muted" sx={{float: 'right'}}>2020-01-01</Box>
          </ListGroup.Item>
          <ListGroup.Item className={cls({borderBottom: 0})}>
            <a href="#" target="_blank">Vestibulum at eros</a>
            <Box display="inline" color="muted" sx={{float: 'right'}}>2020-01-01</Box>
          </ListGroup.Item>
        </ListGroup>
      </Box>
    </Page>;
  }
}
