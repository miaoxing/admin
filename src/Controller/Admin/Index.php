<?php

namespace Miaoxing\Admin\Controller\Admin;

class Index extends \Miaoxing\Plugin\BaseController
{
    protected $controllerName = '后台首页';

    protected $actionPermissions = [
        'index' => '首页',
    ];

    /**
     * 后台首页
     */
    public function indexAction($req)
    {
        return $this->response->redirect($this->url('admin/article'));
    }
}
