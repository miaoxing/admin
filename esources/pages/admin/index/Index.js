import React from "react";
import {Card, Col, ListGroup, Row} from "react-bootstrap";
import styled from 'styled-components';
import app from 'app';
import color from 'color';
import {Page} from "@miaoxing/page";
import {
  Chart,
  Geom,
  Axis,
  Tooltip,
  Legend,
} from "bizcharts";

const GradientCard = styled(Card)`
   background: linear-gradient(
    135deg,
    ${props => props.color},
    ${props => color(props.color).lighten(0.2).string()}
  );
`;

const CardIcon = styled.i`
  opacity: .15;
  position: absolute;
  top: 50%;
  transform:translateY(-50%);
  right: 2rem;
  font-size: 5rem;
`;

export default class extends React.Component {
  state = {
    charts: [],
    stats: []
  }

  componentDidMount() {
    $.get(app.url('admin-api/index/index'), {loading: true})
      .then(ret => this.setState(ret));
  }

  render() {
    return <Page
      raw
      breadcrumb={false}
    >
      <Card>
        <Card.Header>
          数据统计
        </Card.Header>
        <Card.Body>
          <Row>
            {this.state.stats.map(stat => (
              <Col key={stat.title} lg={3} md={6} sm={12}>
                <GradientCard color={stat.color} text="white" className="border-0">
                  <Card.Body className="font-weight-light">
                    <Card.Title className="font-weight-light">{stat.title}</Card.Title>
                    <Card.Text style={{fontSize: '3em'}} className="mb-0">
                      {stat.value}
                    </Card.Text>
                    <CardIcon className={stat.icon}/>
                  </Card.Body>
                </GradientCard>
              </Col>
            ))}
          </Row>
        </Card.Body>
      </Card>

      <Card className="mt-4">
        <Card.Header>
          交易走势
        </Card.Header>
        <Card.Body>
          <Chart
            height={400}
            data={this.state.charts}
            scale={{
              date: {
                type: 'time',
                mask: 'MM-DD'
              }
            }}
            forceFit
          >
            <Legend/>
            <Axis name="date"/>
            <Axis name="value"/>
            <Tooltip
              crosshairs={{
                type: "y"
              }}
            />
            <Geom
              type="line"
              position="date*value"
              size={2}
              color={"name"}
              shape={"smooth"}
            />
            <Geom
              type="point"
              position="date*value"
              size={4}
              shape={"circle"}
              color={"name"}
              style={{
                stroke: "#fff",
                lineWidth: 1
              }}
            />
          </Chart>
        </Card.Body>
      </Card>

      <Card className="mt-4">
        <Card.Header>
          更新日志
          <a href="#" target="_blank" className="float-right">
            更多{' '}<i className="fa fa-angle-right"/>
          </a>
        </Card.Header>
        <ListGroup variant="flush">
          <ListGroup.Item className="border-bottom-0">
            <a href="#" target="_blank">Cras justo odio</a>
            <span className="float-right text-muted">2020-01-01</span>
          </ListGroup.Item>
          <ListGroup.Item className="border-bottom-0">
            <a href="#" target="_blank">Dapibus ac facilisis in</a>
            <span className="float-right text-muted">2020-01-01</span>
          </ListGroup.Item>
          <ListGroup.Item className="border-bottom-0">
            <a href="#" target="_blank">Vestibulum at eros</a>
            <span className="float-right text-muted">2020-01-01</span>
          </ListGroup.Item>
        </ListGroup>
      </Card>
    </Page>;
  }
}
