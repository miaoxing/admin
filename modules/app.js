import { createRoot } from 'react-dom/client';

import './bootstrap';
import 'antd/dist/reset.css';
import '../scss/index.scss';

import AppContainer from '../components/AppContainer';

const container = document.getElementById('root');
const root = createRoot(container);
root.render(<AppContainer/>);
