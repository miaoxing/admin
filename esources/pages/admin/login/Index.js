import React from "react";
import {Button, Form, Input} from "antd";
import {Box, Heading, Image} from 'rebass/styled-components';
import $ from '@miaoxing/app';
import app from 'plugins/app/resources/modules/app';
import logo from 'plugins/admin/resources/images/logo.png';
import '@miaoxing/icons';

export default class extends React.Component {
  render() {
    return <Box
      width={350}
      mx="auto"
      mt={5}
      p={5}
      bg="white"
    >
      <Box
        mb={4}
        textAlign="center"
      >
        <Image height={30} src={logo}/>
      </Box>
      <Heading
        mb={5}
        textAlign="center"
        fontSize={3}
        fontWeight="normal"
        color="muted"
      >
        登录
      </Heading>
      <Form
        onFinish={async values => {
          const ret = await $.post({
            url: app.url('users/login.json'),
            data: values,
          });
          await $.ret(ret);
          if (ret.code === 1) {
            app.to('admin');
          }
        }}
      >
        <Form.Item
          name="username"
          rules={[{required: true, message: '请输入用户名'}]}
        >
          <Input placeholder="用户名" size="large"/>
        </Form.Item>

        <Form.Item
          name="password"
          rules={[{required: true, message: '请输入密码'}]}
        >
          <Input.Password placeholder="密码" size="large"/>
        </Form.Item>

        <Form.Item>
          <Button type="primary" htmlType="submit" size="large" block>
            登 录
          </Button>
        </Form.Item>
      </Form>
    </Box>;
  }
}
