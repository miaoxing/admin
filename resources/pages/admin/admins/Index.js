import React from 'react';
import app from 'app';
import {Button, FormCheck, OverlayTrigger, Popover} from 'react-bootstrap';
import Page from 'components/Page';
import PageHeader from 'components/PageHeader';
import SearchForm from 'components/SearchForm';
import SearchItem from 'components/SearchItem';
import Options from 'components/Options';
import Table from 'components/Table';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faChevronRight, faQuestionCircle} from '@fortawesome/free-solid-svg-icons';
import TableStatusCheckbox from 'components/TableStatusCheckbox';

export default class extends React.Component {
  render() {
    return (
      <Page>
        <PageHeader>
          <Button variant="success" href={app.curNewUrl()}>添加</Button>
        </PageHeader>

        <SearchForm className="js-admin-form">
          <SearchItem label="用户名" name="username"/>

          <SearchItem label="姓名" name="name"/>

          <SearchItem label="昵称" name="nickName"/>

          <SearchItem label="分组" name="groupId">
            <Options data={wei.groups} labelKey="name" valueKey="id" placeholder="全部"/>
          </SearchItem>
        </SearchForm>

        <Table
          columns={[
            {
              text: '用户名',
              dataField: 'username'
            },
            {
              text: '姓名',
              dataField: 'name'
            },
            {
              text: '昵称',
              dataField: 'nickName'
            },
            {
              text: '分组',
              dataField: 'group.name'
            },
            {
              text: '创建时间',
              dataField: 'createTime'
            },
            {
              text: '启用',
              dataField: 'enable',
              headerFormatter: (column) => {
                return <OverlayTrigger trigger="hover" overlay={<Popover>禁用后,用户将无法登录</Popover>}>
                  <span>启用 <FontAwesomeIcon icon={faQuestionCircle}/></span>
                </OverlayTrigger>
              },
              formatter: (cell, row) => {
                return <TableStatusCheckbox url={app.url('admin/admins/enable')} row={row} name="enable"/>
              }
            },
            {
              text: '操作',
              dataField: 'id',
              formatter: (id) => (
                <a href={app.curEditUrl(id)}>编辑</a>
              ),
            }
          ]}
        />
      </Page>
    )
  }
}
