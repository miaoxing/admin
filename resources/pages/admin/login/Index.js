import React from "react";
import {Button, Form, Input} from "antd";
import {Box, Heading, Image} from 'rebass/styled-components';
import $ from 'miaoxing';
import app from 'plugins/app/resources/modules/app';
import logo from 'plugins/admin/resources/images/logo.png';
import {createGlobalStyle} from 'styled-components';
import '@miaoxing/icons';
import {history} from "@miaoxing/app";

const GlobalStyle = createGlobalStyle`
  body {
    background: #f5f8fa url(https://image-10001577.image.myqcloud.com/uploads/3/20190602/15594729401485.jpg);
    background-size: cover;
  }
`;

export default class extends React.Component {
  render() {
    return <>
      <GlobalStyle/>
      <Box
        width={350}
        mx="auto"
        mt={5}
        p={5}
        bg="white"
      >
        <Box
          mb={4}
          textAlign="center"
        >
          <Image height={30} src={logo}/>
        </Box>
        <Heading
          mb={5}
          textAlign="center"
          fontSize={3}
          fontWeight="normal"
          color="muted"
        >
          登录
        </Heading>
        <Form
          onFinish={async values => {
            const ret = await $.post({
              url: app.url('users/login.json'),
              data: values,
            });
            await $.ret(ret);
            if (ret.code === 1) {
              history.push(app.url('admin'));
            }
          }}
        >
          <Form.Item
            name="username"
            rules={[{required: true, message: '请输入用户名'}]}
          >
            <Input placeholder="用户名" size="large"/>
          </Form.Item>

          <Form.Item
            name="password"
            rules={[{required: true, message: '请输入密码'}]}
          >
            <Input.Password placeholder="密码" size="large"/>
          </Form.Item>

          <Form.Item>
            <Button type="primary" htmlType="submit" size="large" block>
              登 录
            </Button>
          </Form.Item>
        </Form>
      </Box>
    </>;
  }
}
