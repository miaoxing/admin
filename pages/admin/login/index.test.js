import Index from './index';
import $, {Ret} from 'miaoxing';
import {fireEvent, render, waitFor} from '@testing-library/react';
import {bootstrap, resetUrl, setUrl} from '@mxjs/test';
import {app} from '@mxjs/app';

bootstrap();

describe('login', () => {
  beforeEach(() => {
    setUrl('admin/login');
    app.page = {
      collection: 'admin/login',
      index: false,
    };
  });

  afterEach(() => {
    resetUrl();
    app.page = {};
  });

  test('suc', async () => {
    $.http = jest.fn().mockResolvedValueOnce({
      ret: Ret.suc({
        token: 'test',
      }),
    });

    window.localStorage = jest.fn();

    const {container, ...result} = render(<Index/>);

    fireEvent.change(result.getByPlaceholderText('用户名'), {target: {value: 'admin'}});
    fireEvent.change(result.getByPlaceholderText('密码'), {target: {value: 'password'}});
    fireEvent.click(result.getByText('登 录'));

    await waitFor(() => {
      expect($.http).toBeCalled();
    });

    expect($.http).toMatchSnapshot();
    expect(window.localStorage).toMatchSnapshot();
    expect(window.location.href).toBe('/admin');
  });
});
