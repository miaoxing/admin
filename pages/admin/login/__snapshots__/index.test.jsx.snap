// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login err 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "login",
      },
    ],
    Array [
      Object {
        "data": Object {
          "password": "password",
          "username": "admin",
        },
        "method": "post",
        "url": "login",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`login suc 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "login",
      },
    ],
    Array [
      Object {
        "data": Object {
          "password": "password",
          "username": "admin",
        },
        "method": "post",
        "url": "login",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`login suc 2`] = `
Storage {
  "token": "test",
}
`;
